home-application/src/main/java/com/softserveinc/ita/homeproject/application/service/impl/QueryApiServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package com.softserveinc.ita.homeproject.application.service.impl;
+
+import javax.ws.rs.core.UriInfo;
+
+import com.softserveinc.ita.homeproject.application.service.QueryApiService;
+import com.softserveinc.ita.homeproject.homedata.entity.BaseEntity;
+import com.softserveinc.ita.homeproject.homeservice.dto.BaseDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+/**
+ * QueryApiServiceImpl class is implementation of QueryApiService - service
+ * that provides Spring Data Page by request query
+ *
+ * @author Oleksii Zinkevych
+ * @see com.softserveinc.ita.homeproject.application.service.QueryApiService
+ */
+@Service
+@RequiredArgsConstructor
+public class QueryApiServiceImpl<T extends BaseEntity, D extends BaseDto> implements QueryApiService<T, D> {
```

Comment:
Not sure it's worth it to add services and apis for query just to get rid of filling map with query parameters because it just adds more code and makes it more complex which doesn't match with pattern KISS

Reply:
How many times did you make mistake filling this map? I found at least two places. The same can be with any dev. In case when we can avoid any manual routine operation we should automate it =)